backend/geonature/core/gn_synthese/routes.py

```diff
@@ -812,21 +812,24 @@ def get_count_observation(id_dataset):
     return DB.session.query(Synthese.cd_nom).filter(Synthese.id_dataset == id_dataset).count()
 
 
-@routes.route("/repartition_taxons_dataset/<int:id_dataset>", methods=["GET"])
+@routes.route("/dataset_taxa_distribution/<int:id_dataset>", methods=["GET"])
 @json_resp
-def get_repartition_taxons(id_dataset):
+def get_taxa_distribution(id_dataset):
```

Comment:
Il ne faut pas mettre en dur le "id_dataset", ni le where en dur, ni dans le nom de la route. On a dit que cette route était générique et qu'elle pouvait renvoyer le nombre de taxon par année, cadre d'acquisition, jdd, commune etc...
```suggestion
@routes.route("/taxa_number", methods=["GET"])
@json_resp
def get_taxa_number(id_dataset):
    # on récupere les paramètres passer en query string
    params = request.args
    # on construit la requête
    query = DB.session.query(
        func.count(distinct(Synthese.cd_nom)),
        Taxref.regne
    ).select_from(
        Synthese
    ).outerjoin(
        Taxref, Taxref.cd_nom == Synthese.cd_nom
    )
    # et on l'adapte en fonction des paramètre de la query string
    if 'id_dataset' in params:
        query = query.filter(Synthese.id_dataset = params['id_dataset']
    # on rajoutera ensuite d'autres filtres en fonction des paramètres et des besoins
```